src/main/java/org/jmxtrans/agent/util/io/UrlResource.java

```diff
@@ -0,0 +1,178 @@
+import java.io.InputStream;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.net.URLConnection;
+import java.util.concurrent.TimeUnit;
+
+import javax.annotation.Nonnull;
+
+/**
+ * {@link Resource} for {@code http://...}, {@code https://...} or {@code file://...}.
+ *
+ * @author <a href="mailto:cleclerc@cloudbees.com">Cyrille Le Clerc</a>
+ */
+public class UrlResource extends AbstractResource implements Resource {
+    private final URL url;
+    private final URI uri;
+
+    public UrlResource(@Nonnull String url) {
+        Preconditions2.checkNotNull(url, "Given url cannot be null");
+        try {
+            this.url = new URL(url);
+        } catch (MalformedURLException e) {
+            throw new IoRuntimeException("Exception parsing URL '" + url + "'", e);
+        }
+        URI uri;
+        try {
+            uri = this.url.toURI();
+        } catch (URISyntaxException e) {
+            throw new IoRuntimeException("Exception parsing URL '" + url + "'", e);
+        }
+        this.uri = uri;
+    }
+
+
+    @Nonnull
+    @Override
+    public File getFile() throws IOException {
+        if (IoUtils.isFileUrl(url)) {
+            return new File(uri);
+        } else {
+            return super.getFile();
+        }
+    }
+
+    @Override
+    public boolean exists() {
+        if (IoUtils.isFileUrl(url)) {
+            return super.exists();
+        } else {
+            URLConnection conn;
+            try {
+                conn = url.openConnection();
+                if (conn instanceof HttpURLConnection) {
+                    HttpURLConnection httpConn = (HttpURLConnection) conn;
+                    configureUrlConnection(httpConn);
+                    try {
+                        int responseCode = httpConn.getResponseCode();
+                        if (responseCode == HttpURLConnection.HTTP_OK) {
+                            return true;
+                        } else if (responseCode == HttpURLConnection.HTTP_NOT_FOUND) {
+                            return false;
+                        } else if (httpConn.getContentLength() > 0) {
+                            return true;
+                        } else {
+                            return false;
+                        }
+                    } finally {
+                        httpConn.disconnect();
+                    }
+                } else {
+                    return super.exists();
+                }
+            } catch (IOException e) {
+                return false;
+            }
+        }
+    }
+
+    @Override
+    public long lastModified() throws IOException {
+        if (IoUtils.isFileUrl(url)) {
+            return super.lastModified();
+        } else {
+            URLConnection conn = null;
+            try {
+                conn = url.openConnection();
+                configureUrlConnection(conn);
```

Comment:
A minor detail, but I think it would be a good idea to make a utility method to open _and_ configure a connection to minimize the risk of forgetting to configure it if we refactor